packages/data/src/namespace-store.js

```diff
@@ -105,24 +105,28 @@ function createReduxStore( reducer, key, registry ) {
  * @return {Object}           Selectors mapped to the redux store provided.
  */
 function mapSelectors( selectors, store, registry ) {
-	const createStateSelector = ( registeredSelector ) => function runSelector() {
+	const createStateSelector = ( registeredSelector ) => {
 		const selector = registeredSelector.isRegistrySelector ? registeredSelector( registry ) : registeredSelector;
 
-		// This function is an optimized implementation of:
-		//
-		//   selector( store.getState(), ...arguments )
-		//
-		// Where the above would incur an `Array#concat` in its application,
-		// the logic here instead efficiently constructs an arguments array via
-		// direct assignment.
-		const argsLength = arguments.length;
-		const args = new Array( argsLength + 1 );
-		args[ 0 ] = store.getState();
-		for ( let i = 0; i < argsLength; i++ ) {
-			args[ i + 1 ] = arguments[ i ];
+		function runSelector() {
+			// This function is an optimized implementation of:
+			//
+			//   selector( store.getState(), ...arguments )
+			//
+			// Where the above would incur an `Array#concat` in its application,
+			// the logic here instead efficiently constructs an arguments array via
+			// direct assignment.
+			const argsLength = arguments.length;
+			const args = new Array( argsLength + 1 );
+			args[ 0 ] = store.getState();
+			for ( let i = 0; i < argsLength; i++ ) {
+				args[ i + 1 ] = arguments[ i ];
+			}
+
+			return selector( ...args );
 		}
 
-		return selector( ...args );
+		return runSelector;
```

Comment:
minor nitpick here, could you just return the function directly?

Reply:
> could you just return the function directly?

I think we can, but importantly please keep it as named, since it helps provide context in Performance audit results, where it most certainly will make a presence.

e.g. `return function runSelector() {`